snuba/datasets/errors_processor.py

```diff
@@ -109,7 +104,10 @@ def extract_promoted_contexts(
         tags: Mapping[str, Any],
     ) -> None:
         transaction_ctx = contexts.get("trace") or {}
-        if transaction_ctx.get("trace_id", None):
-            output["trace_id"] = str(uuid.UUID(transaction_ctx["trace_id"]))
-        if transaction_ctx.get("span_id", None):
-            output["span_id"] = int(transaction_ctx["span_id"], 16)
+        trace_id = transaction_ctx.get("trace_id", None)
+        span_id = transaction_ctx.get("span_id", None)
+
+        if trace_id:
+            output["trace_id"] = str(uuid.UUID(trace_id))
+        if span_id:
+            output["span_id"] = int(span_id, 16)
```

Comment:
You do not need this change anymore. Up to you.

Reply:
technically it's a performance improvement 😂